test/oxo.js

```diff
@@ -0,0 +1,345 @@
+
+        await asserts.throws(oxo.cancelGame(defaultGameId, {from: player1}))
+        await asserts.throws(oxo.cancelGame(defaultGameId, {from: player2}))
+    });
+
+    it('should cause game to draw', async () => {
+
+        const player1 = accounts[1];
+        const player2 = accounts[2];
+        await oxo.createGame({from: player1, value: defaultBet});
+        await oxo.joinGame(defaultGameId, {from: player2, value: defaultBet});
+        await oxo.move(defaultGameId, 1, {from: player1});
+        await oxo.move(defaultGameId, 5, {from: player2});
+        await oxo.move(defaultGameId, 3, {from: player1});
+        await oxo.move(defaultGameId, 2, {from: player2});
+        await oxo.move(defaultGameId, 8, {from: player1});
+        await oxo.move(defaultGameId, 9, {from: player2});
+        await oxo.move(defaultGameId, 6, {from: player1});
+        await oxo.move(defaultGameId, 4, {from: player2});
+        let result = await oxo.move(defaultGameId, 7, {from: player1});
+
+        assert.equal(result.logs[1].event, 'Draw');
+        assert.equal(result.logs[1].args.gameId.valueOf(), defaultGameId);
+
+
+        assert.equal(result.logs[2].event, 'MoneyTransferred');
+        assert.equal(result.logs[2].args.gameId.valueOf(), defaultGameId);
+        assert.equal(result.logs[2].args.player.valueOf(), player1);
+        assert.equal(result.logs[2].args.amount.valueOf(),  userPayout(defaultBet));
+
+        assert.equal(result.logs[3].event, 'MoneyTransferred');
+        assert.equal(result.logs[3].args.gameId.valueOf(), defaultGameId);
+        assert.equal(result.logs[3].args.player.valueOf(), player2);
+        assert.equal(result.logs[3].args.amount.valueOf(),  userPayout(defaultBet));
+    });
+
+    it('should allow owner to withdraw money', async () => {
+
+        const player1 = accounts[1];
+        const player2 = accounts[2];
+        await oxo.createGame({from: player1, value: defaultBet});
+        await oxo.joinGame(defaultGameId, {from: player2, value: defaultBet});
+        await oxo.cancelGame(defaultGameId, {from: player1});
+        await oxo.cancelGame(defaultGameId, {from: player2});
+        var result = await oxo.withdrawBalance(100, {from: OWNER});
+
+        assert.equal(result.logs[0].event, 'MoneyWithdrawn');
+        assert.equal(result.logs[0].args.addr.valueOf(), OWNER);
+        assert.equal(result.logs[0].args.amount.valueOf(), 100);
+    });
+
+    it('should deny not owner to withdraw money', async () => {
+
+        const player1 = accounts[1];
+        const player2 = accounts[2];
+        await oxo.createGame({from: player1, value: defaultBet});
+        await oxo.joinGame(defaultGameId, {from: player2, value: defaultBet});
+        await oxo.cancelGame(defaultGameId, {from: player1});
+        await oxo.cancelGame(defaultGameId, {from: player2});
+        await asserts.throws(oxo.withdrawBalance(100, {from: player1}));
+    });
+
+
```

Comment:
Should this game allow to create a game twice in a row?

Reply:
why not... ? i thought it is ok to let users create several games and play in the parallel... as long as each game has its own game id

Reply:
I agree. So you probably need to test this possibility. Like if you will make some changes in the contract and this possibility will break you will know it because the test will crush.